app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -156,11 +171,115 @@ private void showReports(List<Report> reports) {
+                        report.put(Report.DATE_KEY, date);
+                        report.put(Report.USER_KEY, ParseUser.getCurrentUser());
+                        report.put(Report.LOCATION_KEY, new ParseGeoPoint(latLng.latitude, latLng.longitude));
+                        report.put(Report.TYPE_OF_CRIME_KEY, crime);
+                        report.saveInBackground(e -> {
+                            if (e != null) {
+                                Log.e(TAG, "showAlertDialogForPoint: exception", e);
+                            } else {
+                                Log.i(TAG, "showAlertDialogForPoint: report saved");
+                            }
+                        });
+                        showMarker(crime.getTag(), latLng.latitude, latLng.longitude, crime.getLevelOfRisk());
+                    }
+                });
+        alertDialog.setButton(DialogInterface.BUTTON_NEGATIVE, "Cancel",
+                (dialog, id) -> dialog.cancel());
+        alertDialog.show();
+    }
+
+    public void populateSpinner(List<TypeOfCrime> crimesList, HashMap<String, TypeOfCrime> crimes, View messageView) {
+        List<String> list = new ArrayList<>();
+        for (TypeOfCrime crime : crimesList) {
+            crimes.put(crime.getTag(), crime);
+        }
+        for (TypeOfCrime crime : crimesList) {
+            list.add(crime.get(TypeOfCrime.TAG_KEY).toString());
+        }
+        Spinner typesOfCrime = messageView.findViewById(R.id.sTypeOfCrime);
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(getContext(), android.R.layout.simple_spinner_item, list);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        typesOfCrime.setAdapter(adapter);
+        typesOfCrime.setSelection(0);
+    }
+
+    private void queryTypeOfCrimes(View messageView, HashMap<String, TypeOfCrime> crimes) {
+        ParseQuery<TypeOfCrime> query = ParseQuery.getQuery(TypeOfCrime.class);
+        query.addDescendingOrder(TypeOfCrime.LEVEL_OF_RISK_KEY);
+        query.findInBackground((crimesList, e) -> {
+            if (e != null) {
+                Log.e(TAG, "Issue with getting crimes", e);
+                return;
+            } else {
+                populateSpinner(crimesList, crimes, messageView);
+            }
+        });
+    }
+
+    private void setupEditDate(View messageView) {
+
+        messageView.findViewById(R.id.sTypeOfCrime);
+        EditText etDate = messageView.findViewById(R.id.etDate);
+        DatePickerDialog.OnDateSetListener onDateSetListener = (view, year, month, dayOfMonth) -> {
+            month = month + 1;
+            String date = month + "/" + dayOfMonth + "/" + year;
+            etDate.setText(date);
+        };
+        LocalDateTime today = LocalDateTime.now();
+        int month = today.getMonth().getValue();
+        int dayOfMonth = today.getDayOfMonth();
+        int year = today.getYear();
+        String date = month + "/" + dayOfMonth + "/" + year;
```

Comment:
You should be able to use a date formatter to do this for you: https://developer.android.com/reference/java/time/format/DateTimeFormatter